experimental/attestation_common/src/keying_material.rs

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2021 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use crate::report::{AttestationInfo, Report};
+use anyhow::{anyhow, Context};
+use serde::{Deserialize, Serialize};
+
+const KEYING_MATERIAL_PURPOSE: &str = "TLSAttestationV1";
+const KEYING_MATERIAL_CLIENT_LABEL: &str =
```

Comment:
Doesn't this influence the keying material? If the client and server use different labels I think they will generate different secrets, which means they do not agree on a shared secret. Or am I missing the point of how this will be used?

Reply:
Correct, labels should be known by both participants in advance and even be registered with IANA.
In this implementation both client and server will export Keying Material with both labels: one to send, and one to check in the response.